synapse/lib/types.py

```diff
@@ -613,6 +615,68 @@ def _normPyTuple(self, valu):
     def indx(self, norm):
         return s_common.buid(norm)
 
+class Hex(Type):
+    _opt_defs = (
+        ('size', 0),
+    )
+
+    def postTypeInit(self):
+        self._size = self.opts.get('size')
+        if self._size < 0:
+            # zero means no width check
+            raise s_exc.BadConfValu(name='size', valu=self._size,
+                                    mesg='Size must be > 0')
+        if self._size % 2 != 0:
+            raise s_exc.BadConfValu(name='size', valu=self._size,
+                                    mesg='Size must be a multiple of 2')
+        self.setNormFunc(str, self._normPyStr)
+        self.setNormFunc(bytes, self._normPyBytes)
+
+    def liftPropEq(self, xact, fenc, penc, valu):
+
+        # Prefix searching is allowed with a '*'
+        if isinstance(valu, str) and valu.endswith('*'):
+            valu = valu.rstrip('*')
+            norm, info = self._normPyStr(valu, chksz=False)
```

Comment:
I'd like to keep the calling conventions for the registered handlers free from one-offs.  Maybe a quick function that just does the lower() and %2 check ( that this *and* normPyStr called ) would clear this up?